job_handlers/azure_container_instances/__init__.py

```diff
@@ -42,7 +42,10 @@ def __init__(self, job, data_source: str):
             client_secret=config.AZURE_JOB_SP_SECRET,
             tenant_id=config.AZURE_JOB_SP_TENANT_ID,
         )
-        self.azure_valid_container_name = self.job.entity["name"].lower().replace(".", "-")
+        if "name" in self.job.entity.keys():
+            self.azure_valid_container_name = self.job.entity["name"].lower().replace(".", "-")
+        else:
+            self.azure_valid_container_name = "azure_container_job"
```

Comment:
"name" does not seem to be specified in the blueprints for "AzureContainer".
So we should ether add it as a required attribute, or remove the need for it here completely.
I'd say add it.